pwiz_tools/Skyline/Model/Proteome/ProteinAssociation.cs

```diff
@@ -190,7 +193,8 @@ public void UseBackgroundProteome(BackgroundProteome backgroundProteome, Func<Fa
             return proteinAssociations;
         }
 
-        public class ParsimonySettings
+        [XmlRoot("protein_association")]
+        public class ParsimonySettings : IXmlSerializable

```

Comment:
As Brendan mentioned, this class needs to be Immutable.
Also, I would recommend that you add a member:
public static readonly ParsimonySettings EMPTY = ...;
that you use everywhere, instead of allowing PeptideSettings.ProteinAssociationSettings to ever be null.

When you are writing out the XML, you can, in PeptideSettings.WriteXml, check "if (!ParsimonySettings.EMPTY.Equals(peptideSettings.ProteinAssociationSettings))" when you are deciding whether any XML needs to be written to the .sky file.

Reply:
Partially done. I assume private set is ok?